src/main/kotlin/org/rust/ide/intentions/IfLetToMatchIntention.kt

```diff
@@ -0,0 +1,118 @@
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.parentOfType
+
+class IfLetToMatchIntention : RsElementBaseIntentionAction<IfLetToMatchIntention.Context>() {
+    override fun getText(): String = "Convert if let statement to match"
+    override fun getFamilyName(): String = text
+
+    data class Context(
+        val ifStmt: RsIfExpr,
+        val target: RsExpr,
+        val matchArms: MutableList<MatchArm>,
+        var elseBody : RsBlock?
+    )
+
+    data class MatchArm(
+        val matchArm: RsPat,
+        val body: RsBlock
+    )
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): Context? {
+        //1) Check that we have an if statement
+        var ifStatement = element.parentOfType<RsIfExpr>() ?: return null
+
+        // We go up in the tree to detect cases like `... else if let Some(value) = x { ... }`
+        // and select the correct if statement
+
+        while(ifStatement.parent is RsElseBranch) {
+            // In that case
+            // typeof(if.parent) = RsElseBranch ==> typeof(if.parent.parent) = RsIfExpr
+            ifStatement = ifStatement.parent.parent as? RsIfExpr ?: return null
+        }
+
+        //Here we have extracted the upper most if statement node
+
+        return extractIfLetStatementIfAny(ifStatement)
+    }
+
+    override fun invoke(project: Project, editor: Editor, ctx: Context) {
+        val (ifStmt, target, matchArms, elseBody) = ctx
+
+        var generatedCode = "match ${target.text} {"
```

Comment:
You can use `buildString` for more efficient and idiomatic code:

```
        buildString { 
            append("foo")
            append("bar")
        }
```

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/build-string.html